ArduCopter/mode_auto.cpp

```diff
@@ -131,8 +131,21 @@ void ModeAuto::run()
         break;
 
     case SubMode::CIRCLE:
-        circle_run();
-        break;
+        // check in which state are we in
+        switch (loiter_turns.get_state()) {
+        case _LoiterTurns::State::CIRCLING: {
+            loiter_turns.circle_run();
+            break;
+        }
+        case _LoiterTurns::State::MOVING_TO_EDGE:
+            // use waypoint navigation controller in this state
+            wp_run();
+            break;
+        default:
+            // no need to do anything for other states
```

Comment:
Name the states here rather than using `default:`

Reasoning for this is that if a state gets added the programmer should think about what to do.